streamer/src/nonblocking/quic.rs

```diff
@@ -354,6 +395,21 @@ async fn run_server(
                 continue;
             }
 
+            let open_connections = stats.open_connections.load(Ordering::Relaxed);
```

Comment:
I think that this is racy? Two threads could load and find open_connections to be max_concurrent_connections - 1.

I think you need to fetch_add and then fetch_sub if the value returned by fetch_add is > max_concurrent_connections

Reply:
fixed